backend/src/core/middlewares/campaign.middleware.ts

```diff
@@ -102,7 +102,10 @@ const listCampaigns = async (
   res: Response,
   next: NextFunction
 ): Promise<Response | void> => {
-  const { offset, limit } = req.query
+  const { offset, limit } = (req.query as unknown) as {
```

Comment:
This is a workaround for the change in type definitions of `Request.query` (info: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43434#issuecomment-607181516).

I also tried defining the type of `Request.query` using generics like so: `req: Request<{}, any, any, { offset: number; limit: number }>`. But this causes the type error to propagate up to the routes.

After defining the type for `routes.get('/campaigns')`, the `celebrate` library errors since it doesn't have generics support. We could upgrade to v13.0.4 which adds generics (https://github.com/arb/celebrate/commit/69a9b973860edb0e243467a7970353d74fa2c5e8), but that would be a major version upgrade and might require more looking into.

Reply:
I'm more inclined to put off this upgrade first in today's release and fix it properly rather than casting `req.query` to `unknown`.

Reply:
Alright, sounds good!